src/client/providers/setInterpreterProvider.ts

```diff
@@ -75,21 +92,36 @@ function suggestPythonPaths(): Promise<vscode.QuickPickItem[]> {
 }
 
 function setPythonPath(pythonPath: string) {
-    // Waiting on https://github.com/Microsoft/vscode/issues/1396
-    // For now, just let the user copy this to clipboard
-    const copy_msg =  "Copy to Clipboard"
-
-    // If the user already has .vscode/settings.json in the workspace
-    // open it for them
     vscode.workspace.openTextDocument(workspaceSettingsPath())
-        .then(doc => vscode.window.showTextDocument(doc));
-    
-    vscode.window.showInformationMessage(pythonPath, copy_msg)
-        .then(item => {
-            if (item === copy_msg) {
-                ncp.copy(pythonPath)
+        .then(doc => {
+            const settingsText = doc.getText();
+            if (settingsText.search(REPLACE_PYTHONPATH_REGEXP) === -1) {
+                // Can't find the setting to replace - will just have to offer a copy button and instruct them to edit themselves.
+                openWorkspaceSettings().then(() => {
+                    const copyMsg =  "Copy to Clipboard"
+                    const newEntry = `"python.pythonPath": "${pythonPath}"`;
+                    vscode.window.showInformationMessage(`Please add an entry: ${newEntry}`, copyMsg)
+                        .then(item => {
+                            if (item === copyMsg) {
+                                ncp.copy(newEntry)
+                            }
+                        })
+                })
+            } else {
+                // Great, the user already has a setting stated that we can relibly replace!
+                const newSettingsText = settingsText.replace(REPLACE_PYTHONPATH_REGEXP, `$1"${pythonPath}"`);
+                replaceContentsOfFile(doc, newSettingsText).then(
+                    () => {
+                        vscode.window.setStatusBarMessage(`Workspace Interpreter set to ${pythonPath}`);
+                        // As the file is saved the following should be the same as each other but they
+                        // aren't - some form of race condition?
+                        // const currentPythonPath = settings.PythonSettings.getInstance().pythonPath;
+                        // console.log(currentPythonPath);
+                        // console.log(pythonPath);
```

Comment:
@DonJayamanne I've left this comment here as it is my only concern with this PR - it feels like this should print two identical paths, but at least in the instant after saving this is not the case.

If you are happy that this makes sense (that the `PythonSettings` will be getting a callback shortly after the save and everything is OK) then we can remove these comments.